src/gameengine/graphics/Renderer.java

```diff
@@ -0,0 +1,355 @@
+package gameengine.graphics;
+
+import gameengine.geometry.Vector2D;
+
+import java.awt.*;
+
+/**
+ * The Renderer is responsible for drawing on the current frame.
+ */
+public abstract class Renderer {
+    /**
+     * Sets the foreground color.
+     *
+     * @param color The foreground color
+     */
+    public final void setForegroundColor(MutableColor color) {
```

Comment:
Sure, I'm open to suggestions but I can't think of anything better at the moment.  

OpenGL took it one step further in complexity so that you set the clear color and explicitly clear the screen (clearing can be limited to a portion of the screen so you don't re-render portions of the screen that haven't changed for improved performance).

Essentially, I'm trying to represent the color that would be displayed on all pixels unless something is drawn over top.  Requiring the user to draw a rectangle over top of each frame is not user friendly and has some difficulties because of ordering.  We could make it more complicated by replacing the render method in the context class with 2 methods renderFirst & renderLast so that render first would be called first (and you could draw the rectangle to cover the screen) after which the entities in the world (that are visible on screen) would be drawn after which the renderLast would be called.  But... there are more complications because of the viewport translation and / or scaling so it's not clear what the dimensions of the rectangle need to be.